docs/UserGuide.md

```diff
@@ -126,118 +177,300 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
+
+### Setting mastery check result for a student : `mc`
+
+Sets mastery check result for a student.
+
+Format: `mc INDEX num/MC_NUM r/MC_RESULT`
+
+* Sets mastery check result for the student at the specified `INDEX`. The index refers to the index number shown in the displayed student list.
+* `INDEX` **must be a positive integer** 1, 2, 3, …​
+* `MC_NUM` must be 1 or 2.
+* `MC_RESULT` must be "pass" or "fail".
+* `MC_RESULT` is not case-sensitive.
+
+Examples:
+* `mc 1 num/1 r/PASS` sets the mastery check result for the 1st student to PASS.
+
+## Task features
+This section contains features for managing tasks.
+
+### Adding a task: `addtask`
+
+Adds a task to JARVIS.
+
+Format: `addtask t/TASK_DESC [d/DEADLINE]`
+
+* `d/DEADLINE` is optional, if specified will indicate a deadline for the task.
+* `d/DEADLINE` should be in the format `yyyy-MM-dd`
+
+Examples:
+* `addtask t/Prepare tutorial slides d/2022-11-01`
+* `addtask t/Mark missions`
+
+### Deleting a task : `deletetask`
+
+Deletes a task from the task list.
+
+Format: `deletetask INDEX`
+
+* `INDEX` refer to the index number shown in the task list.
+* `INDEX` **must be a positive integer** 1, 2, 3, …​
+
+Examples:
+* `deletetask 2` deletes the 2nd task in JARVIS.
+
+### Marking a task as done : `marktask`
 
 Marks a task as done.
 
-Format: `mark t/TASK_INDEX`
+Format: `marktask INDEX`
 
-**Things to Note:**
-* `TASK_INDEX` refers to the index number shown in the displayed list of tasks.
-* `TASK_INDEX` **must be a positive integer** 1, 2, 3, …​
+* `INDEX` refers to the index number shown in the displayed list of tasks.
+* `INDEX` **must be a positive integer** 1, 2, 3, …​
 
 Example:
-* `mark t/2` marks the 2nd task as done.
+* `marktask 2` marks the 2nd task as done.
 
-### Marking a task as not done : `unmark`
+### Marking a task as not done : `unmarktask`
 
 Marks a task as not done.
 
-Format: `unmark t/TASK_INDEX`
+Format: `unmarktask INDEX`
 
-**Things to Note:**
-* `TASK_INDEX` refers to the index number shown in the displayed list of tasks.
-* `TASK_INDEX` **must be a positive integer** 1, 2, 3, …​
+* `INDEX` refers to the index number shown in the displayed list of tasks.
+* `INDEX` **must be a positive integer** 1, 2, 3, …​
 
 Example:
-* `unmark t/2` marks the 2nd task as not done.
+* `unmarktask 2` marks the 2nd task as not done.
+
+## Lesson features
+This section contains features for managing lessons.
+
+### Adding a studio lesson: `addstudio`
+
+Adds a studio lesson to JARVIS.
 
-### Deleting a student/task : `delete`
+Format: `addstudio [l/LESSON_DESC] sd/START_DATE st/START_TIME [ed/END_DATE] et/END_TIME`
 
-Deletes the specified student or task from their respective lists.
+* `l/LESSON_DESC` is optional
+* `sd/START_DATE` and `ed/END_DATE` should be in the format `yyyy-MM-dd`.
+* `st/START_TIME` and `et/END_TIME` should be in the format `hh:mm`.
+* `ed/END_DATE` is optional, if unspecified `END_DATE` will be assumed to be the same as START_DATE
```

Comment:
Might be better to use inline code for START_DATE to be consistent i.e. change to `START_DATE`